consul/src/main/scala/io/buoyant/consul/v1.scala

```diff
@@ -121,6 +126,128 @@ package object v1 {
     }
   }
 
+  object KvApi {
+    def apply(c: Client): KvApi = new KvApi(c, s"/$versionString")
+  }
+
+  class KvApi(
+    client: Client,
+    uriPrefix: String,
+    backoffs: Stream[Duration] = Backoff.exponentialJittered(1.milliseconds, 5.seconds),
+    stats: StatsReceiver = DefaultStatsReceiver
+  ) extends Closable {
+    def close(deadline: Time) = client.close(deadline)
+
+    val kvPrefix = s"$uriPrefix/kv"
+
+    private[this] val infiniteRetryFilter = new RetryFilter[http.Request, http.Response](
+      RetryPolicy.backoff(backoffs) {
+        // We will assume 5xx are retryable, everything else is not for now
+        case (_, Return(rep)) => rep.status.code >= 500 && rep.status.code < 600
+        case (_, Throw(NonFatal(ex))) =>
+          log.error(s"retrying consul kv request on error $ex")
+          true
+      },
+      HighResTimer.Default,
+      stats,
+      RetryBudget.Infinite
+    )
+
+    def retryClient(retry: Boolean) = {
+      val retryFilter = if (retry) infiniteRetryFilter else Filter.identity[http.Request, http.Response]
+      retryFilter andThen client
+    }
+
+    // https://www.consul.io/docs/agent/http/kv.html#single
+    def list(
+      path: String,
```

Comment:
might be worth noting in a comment/scaladoc that path must begin with a slash (I assume?)